### PR TITLE
Handle frame projection when there is a large jump/dropout in packets.

### PR DESCRIPTION
**Pull request by JonathanLennox**
Note this incorporates https://github.com/jitsi/jitsi-videobridge/pull/1081 as well.

### CODE REVIEW

File: src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java [hunk 1]
```diff
@@ -94,7 +94,7 @@
     /**
      * Whether to enable or disable book keeping.
      */
-    public static final Boolean ENABLE_BOOKKEEPING = false;
+    public static final Boolean ENABLE_BOOKKEEPING = true;
```
Comment (bgrozev) [inline src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java]: Is this intentional?

Comment (JonathanLennox) [reply src/main/java/org/jitsi/videobridge/util/ByteBufferPool.java]: Whoops, no, thanks.

File: src/main/java/org/jitsi/videobridge/cc/vp8/VP8FrameMap.java [hunk 2]
```diff
@@ -367,6 +438,20 @@ public int interpret(int seq)
...
+            /** Force this sequence to be interpreted as the new highest, regardless
+             * of its rollover state.
+             */
+            public int resetAt(int seq)
+            {
+                int delta = Vp8Utils.getExtendedPictureIdDelta(seq, highestSeqNumReceived);
+                if (delta < 0)
+                {
+                    roc++;
+                    highestSeqNumReceived = seq;
```
Comment (bgrozev) [inline src/main/java/org/jitsi/videobridge/cc/vp8/VP8FrameMap.java]: Should this also happen when `delta >= 0`?

File: src/main/java/org/jitsi/videobridge/cc/vp8/VP8FrameMap.java [hunk 3]
```diff
@@ -196,6 +252,12 @@ public boolean isNewFrame()
         {
             return newFrame;
         }
+
+        /** Get whether inserting the frame created a new frame. */
```
Comment (bgrozev) [inline src/main/java/org/jitsi/videobridge/cc/vp8/VP8FrameMap.java]: s/created a new frame/caused a reset/?

File: src/main/java/org/jitsi/videobridge/cc/vp8/VP8FrameMap.java [hunk 4]
```diff
@@ -75,9 +76,47 @@ private FrameInsertionResult doFrameInsert(@NotNull VP8Frame frame, Vp8Packet pa
             logger.warn(e);
         }
         frame.addPacket(packet);
-        return new FrameInsertionResult(frame, false);
+        return new FrameInsertionResult(frame, false, false);
     }
 
+    /** Check whether this is a large jump from previous state; if so, reset the map. */
```
Comment (bgrozev) [inline src/main/java/org/jitsi/videobridge/cc/vp8/VP8FrameMap.java]: The way this is phrased makes me expect the method to do the reset, but it doesn't.

File: src/main/java/org/jitsi/videobridge/cc/vp8/VP8FrameMap.java (hunk 2 above)
Comment (bgrozev) [reply src/main/java/org/jitsi/videobridge/cc/vp8/VP8FrameMap.java]: Nevermind, I see this happens in `getIndex()`

**Comment by bgrozev** [comment]
I don't understand it fully, but LGTM (left minor comments)

**Pull request merged**